### DISCUSSION TITLE
Git Discussion

### DISCUSSION THREAD

**Discussion by Jerry2272** [community/community#34555] [New to GitHub]
Someone please I need help with these Git and Github, how does it really work because I find it so difficult to understand..

**Comment by thaisonzdx** [comment]
Git is a **software**, stored your project **history** (on your local computer). And Github is a server had been installed Git, you can **mirror** your local project to it.

To save a state (history) of your code. You can use 2 command `git add <file(s)>` and `git commit`. This is an example about the history (chain of history) of the **a.txt** file

**State 1:**
```
Hello, World
```

**State 2:**
```
Hello, World
Hi, Github
```

**State 3:**
```
Hello,
Hi, Github
```
and so on...

States had been saved seperately. By the way, Git helps you see **where code** had been inserted/deleted (indentify by **line number**) and **who** is author of this changes. Of course, it not only track a file, but also track **whole files** on your project dir.

You can see more in git references: [https://git-scm.com/book/en/v2](https://git-scm.com/book/en/v2)

**Reply by Jerry2272** [reply]
Thanks Sownt. If you don't can I have your Facebook page so i can add you up we can as well video chat so that you can teach me these git and GitHub to my understanding. Please as am new here into these Git and Github Stuff I just need you to get me through so I can be able to be posting my repository please assist me on these and I will forever be grateful to you. Thanks

**Comment by wizardigor** [comment]
Esse canal é muito bom para quem esta iniciando ou tem duvidas so bre o GitHub. Muito recomendado.
[Rafaella Ballerini](https://www.youtube.com/c/rafaellaballerini)

**Reply by Jerry2272** [reply]
Hey what's up Wizardigor, how are you doing
Are you on Facebook?

**Reply by Jerry2272** [reply]
So tell me how do I go about about with it.

**Reply by wizardigor** [reply]
é bem intuitivo, basta seguir os passos.

**Reply by Jerry2272** [reply]
please are you o Facebook?

**Comment by jfullstackdev** [comment]
this one 

https://www.youtube.com/watch?v=RGOj5yH7evk

**Comment by airtower-luna** [comment]
Check the ["Git Book" website](https://git-scm.com/book/en/v2). The first three chapters explain the basics. After that it's a bit pick and choose, up to the technical details if you want. :slightly_smiling_face:

**Comment by lucasrmagalhaes** [comment]
Hey! Check the [Skills GitHub](https://skills.github.com/ ).

**Comment by vicvenpet** [comment]
Hey buddy! Github and Git are very useful and necessary tools to be able to perform activities on the platform frequently, I recommend you to read the Github [Git Guides](https://github.com/git-guides) and [About Git](https://docs.github.com/es/get-started/using-git/about-git) to improve in the open-source world, greetings! 🤟

**Reply by Jerry2272** [reply]
Thanks guys.. for  all your contributions.. Is any of you here on Facebook?

**Comment by cia1099** [comment]
You should watch tutorial video. There are many videos in YouTube and MIT courses

**Comment by GeauxWeisbeck4** [comment]
I love GitHub - I know it inside out so if you need help, let me know!

**Reply by Jerry2272** [reply]
Yes of course please I need help..

**Reply by GeauxWeisbeck4** [reply]
Email me at [redacted due to PII] or hit me up on Discord - geauxweisbeck4#9944 I also have a channel that offers web dev help https://discord.gg/GprTpHGz

**Reply by Jerry2272** [reply]
Can I have your WhatsApp number please

**Comment by dr5hn** [comment]
One more interactive way to learn github https://ohmygit.org/
🥂
